src/download.js

```diff
@@ -1,124 +1,94 @@
 'use strict';
 
-var os = require('os');
 var path = require('path');
-var fs = require('fs');
-var mkdirp = require('mkdirp');
-var GitHub = require('github-releases-ms');
-var ProgressBar = require('progress');
+const { downloadArtifact } = require('@electron/get');
+const { getDownloadUrl } = require('./util');
+const ProgressBar = require('progress');
 var semver = require('semver');
 var rename = require('gulp-rename');
 var es = require('event-stream');
 var zfs = require('gulp-vinyl-zip');
 var filter = require('gulp-filter');
 var assign = require('object-assign');
 
-var cachePath = path.join(os.tmpdir(), 'gulp-electron-cache');
-
-function cache(assetName, repo, onMiss, cb) {
-	var assetFolder = path.join(cachePath, repo);
-	mkdirp(assetFolder, function (err) {
-		if (err) { return cb(err); }
-
-		var assetPath = path.join(assetFolder, assetName);
-
-		fs.exists(assetPath, function (exists) {
-			if (exists) { return cb(null, assetPath); }
-
-			var tempAssetPath = assetPath + '.tmp';
-			onMiss(tempAssetPath, function (err) {
-				if (err) { return cb(err); }
-
-				fs.rename(tempAssetPath, assetPath, function (err) {
-					if (err) { return cb(err); }
-
-					cb(null, assetPath);
-				});
-			});
-		});
-	});
-}
-
 function download(opts, cb) {
-	var repo = opts.repo || 'atom/electron';
-	var github = new GitHub({ repo: repo, token: opts.token });
+	let bar;
 
 	if (!opts.version) {
 		return cb(new Error('Missing version'));
 	}
 
-	var platform = opts.platform;
-	if (!platform) {
+	if (!opts.platform) {
 		return cb(new Error('Missing platform'));
 	}
 
-	var arch = opts.arch;
-
+	let arch = opts.arch;
 	if (!arch) {
-		switch (platform) {
+		switch (opts.platform) {
 			case 'darwin': arch = 'x64'; break;
 			case 'win32': arch = 'ia32'; break;
 			case 'linux': arch = 'ia32'; break;
 		}
 	}
 
-	var version = 'v' + opts.version;
-	var assetName = [opts.assetName, version, platform, arch].join('-') + '.zip';
-
-	function download(assetPath, cb) {
-		github.getReleases({ tag_name: version }, function (err, releases) {
-			if (err) { return cb(err); }
-
-			var release = releases[0];
-
-			if (!release) {
-				return cb(new Error('No release ' + opts.version + ' found'));
-			}
-
-			var asset = release.assets.filter(function (asset) {
-				return asset.name === assetName;
-			})[0];
-
-			if (!asset) {
-				return cb(new Error('No asset for version ' + opts.version + ', platform ' + platform + ' and arch ' + arch + ' found'));
-			}
+	const downloadOpts = {
+		version: opts.version,
+		platform: opts.platform,
+		arch,
+		token: opts.token
```

Comment:
Also needs `artifactName: opts.assetName ? opts.assetName : 'electron'` for the default case.

Otherwise results in `Error: Expected property "artifactName" to be provided as a string but it was not`